hpedockerplugin/etcdutil.py

```diff
@@ -503,40 +503,36 @@ def get_backend_key(self, backend):
 
 
 class EtcdLock(object):
-    # To use this class with "with" clause, passing
-    # name is MUST
-    def __init__(self, lock_root, client, name=None):
+    def __init__(self, lock_root, client, name):
         self._lock_root = lock_root
         self._client = client
         self._name = name
+        self._lock = etcd.Lock(client, name)
 
     def __enter__(self):

```

Comment:
Is this for use of the with clause of statements.. where the lock acquire/release will be done automatically ? More of a clarification for my understanding.

Reply:
Yes William. Implementation of __enter__() and __exit__() allows caller to use instance of this class to be used with 'with' clause.